escrow/src/main.sw

```diff
@@ -1,257 +1,355 @@
-        // require(storage.state == State::Void, Error::CannotReinitialize);
-        require(storage.state == 0, Error::CannotReinitialize);
-
-        storage.asset_amount = asset_amount;
-        storage.buyer = User {
-            address: buyer, approved: false, deposited: false
-        };
-        storage.seller = User {
-            address: seller, approved: false, deposited: false
-        };
-        storage.asset = asset;
-        storage.state = 1;
-        // storage.state = State::Pending;
+        // We must reset the proposal or the escrow contract will be drained
+        storage.arbiter_proposal.insert(identifier, Option::None);
+        storage.escrows.insert(identifier, escrow);
 
-        true
+        log(AcceptedArbiterEvent {
+            identifier
+        });
     }
 
-    /// Updates the user state to indicate that they have deposited
-    /// A successful deposit unlocks the approval functionality
-    ///
-    /// # Panics
-    ///
-    /// The function will panic when
-    /// - The constructor has not been called to initialize
-    /// - The user is not an authorized user that has been set in the constructor
-    /// - The user deposits an asset that is not the specified asset in the constructor
-    /// - The user sends an incorrect amount of the asset that has been specified in the constructor
-    /// - The user deposits when they still have their previous deposit in the escrow
-    #[storage(read, write)]fn deposit() -> bool {
-        // require(storage.state == State::Pending, Error::StateNotPending);
-        require(storage.state == 1, Error::StateNotPending);
-        require(storage.asset == msg_asset_id(), Error::IncorrectAssetId);
-        require(storage.asset_amount == msg_amount(), Error::IncorrectAssetAmount);
-
-        let sender: Result<Identity, AuthError> = msg_sender();
-
-        if let Identity::Address(address) = sender.unwrap() {
-            require(address == storage.buyer.address || address == storage.seller.address, Error::UnauthorizedUser);
-
-            if address == storage.buyer.address {
-                require(!storage.buyer.deposited, Error::UserHasAlreadyDeposited);
-
-                storage.buyer.deposited = true;
-            } else if address == storage.seller.address {
-                require(!storage.seller.deposited, Error::UserHasAlreadyDeposited);
-
-                storage.seller.deposited = true;
-            }
-        } else {
-            revert(0);
+    #[storage(read, write)]fn create_escrow(arbiter: Arbiter, assets: [Asset;
+    2], buyer: Identity, deadline: u64) {
+        // The assertions ensure that assets are specified with a none-zero amount, the arbiter is
+        // not the buyer / seller, the arbiter has a fee that they can take upon resolving a dispute
+        // and the escrow deadline is set in the future
+
+        // require(0 < assets.len(), AssetInputError::UnspecifiedAssets);
```

Comment:
I would add a general comment somewhere to this file explaining that the commented lines will be uncommented once the SDK supports vecs.